aa_stripe/models.py

```diff
@@ -58,6 +59,89 @@ class Meta:
+    DURATION_ONCE = "once"
+    DURATION_REPEATING = "repeating"
+    DURATION_CHOICES = (
+        (DURATION_FOREVER, DURATION_FOREVER),
+        (DURATION_ONCE, DURATION_ONCE),
+        (DURATION_REPEATING, DURATION_REPEATING)
+    )
+
+    coupon_id = models.CharField(max_length=255, help_text=_("Identifier for the coupon"))
+    amount_off = models.PositiveIntegerField(
+        blank=True, null=True, help_text=_("Amount (in the currency specified) that will be taken off the subtotal of"
+                                           "any invoices for this customer."))
+    currency = models.CharField(
+        max_length=3, default="usd", help_text=_("If amount_off has been set, the three-letter ISO code for the"
+                                                 "currency of the amount to take off."))
+    duration = models.CharField(
+        max_length=255, choices=DURATION_CHOICES,
+        help_text=_("Describes how long a customer who applies this coupon will get the discount."))
+    duration_in_months = models.PositiveIntegerField(
+        blank=True, null=True, help_text=_("If duration is repeating, the number of months the coupon applies."
+                                           "Null if coupon duration is forever or once."))
+    livemode = models.BooleanField(
+        default=False, help_text=_("Flag indicating whether the object exists in live mode or test mode."))
+    max_redemptions = models.PositiveIntegerField(
+        blank=True, null=True,
+        help_text=_("Maximum number of times this coupon can be redeemed, in total, before it is no longer valid."))
+    metadata = JSONField(help_text=_("Set of key/value pairs that you can attach to an object. It can be useful for"
+                                     "storing additional information about the object in a structured format."))
+    percent_off = models.PositiveIntegerField(
+        blank=True, null=True,
+        help_text=_("Percent that will be taken off the subtotal of any invoicesfor this customer for the duration of"
+                    "the coupon. For example, a coupon with percent_off of 50 will make a $100 invoice $50 instead."))
+    redeem_by = models.DateTimeField(
+        blank=True, null=True, help_text=_("Date after which the coupon can no longer be redeemed."))
+    times_redeemed = models.PositiveIntegerField(
+        default=0, help_text=_("Number of times this coupon has been applied to a customer."))
+    valid = models.BooleanField(
+        default=False,
+        help_text=_("Taking account of the above properties, whether this coupon can still be applied to a customer."))
+    created = models.DateTimeField()
+    is_deleted = models.BooleanField(default=False)
+
+    def save(self, *args, **kwargs):
+        stripe.api_key = settings.STRIPE_API_KEY
+        if self.pk:
+            try:
+                coupon = stripe.Coupon.retrieve(self.coupon_id)
+                coupon.metadata = self.metadata
+                coupon.save()
```

Comment:
You have super below which should take care of save, right?
You can also check changed fields like here: https://github.com/ArabellaTech/xcheck_api/blob/develop/xcheck_api/campaigns/models.py#L259